lua/various-textobjs/charwise-textobjs.lua

```diff
@@ -351,6 +351,32 @@ function M.mdlink(scope, lookForwL)
 	selectTextobj(pattern, scope, lookForwL)
 end
 
+---@param scope "inner"|"outer" inner selector only includes the content, outer selector includes the type.
+---@param lookForwL integer
+function M.mdEmphasis(scope, lookForwL)
+	local patterns = {}
+	for _, tag in ipairs({ "*", "**", "***", "_", "__", "___" }) do
+		local tagChar = tag:sub(1, 1)
+		local escapedTag = tag
+		if tagChar == "*" then
+			escapedTag = tag:gsub("%*", "%%*")
+		end
+		table.insert(patterns, ("^(%s)[^%s](%s)"):format(escapedTag, tagChar, escapedTag))
+		table.insert(patterns, ("^(%s)[^%s].-[^\\](%s)"):format(escapedTag, tagChar, escapedTag))
+		table.insert(patterns, ("([^\\%s]%s)[^%s](%s)"):format(tagChar, escapedTag, tagChar, escapedTag))
```

Comment:
just a tip: you can simplify `[^%s]` to `%S`

Reply:
`[^%s]` is actually a formatter string, but it can't include spaces anyway. Updated.